Added resizing to screen_events.c (#1741)

tlubke opened the pull request.

tlubke commented:
Might make a separate PR later for dropping inferred "frames" instead of clearing the whole queue.

tehn reviewed (approved):
Code looks good, and curious about the use case but happy to merge.

tlubke commented:
Originally closed because I had totally disregarded the queue was a circular buffer... So, I fixed the resize logic and reworked the reset function a little.

For anyone reviewing, please keep a skeptical eye open for corner cases (I feel like I might be missing something).

tehn commented:
Code looks good with my quick review, did not test on device.

Given the amount of RAM this system has, would it be bad to just keep the queue always really big?

tlubke commented:
> Given the amount of RAM this system has, would it be bad to just keep the queue always really big?

Certainly could take this approach, each `screen_event_data` should be 64-bytes assuming worst case alignment. Or 0.5MiB if `SCREEN_Q_SIZE` is changed to 8192.

tehn commented:
Given the processor has 1GB of RAM, I feel the tradeoff for simply allocating the relatively small amount of memory would be a good tradeoff to the extra complexity for changing the buffer size.

tlubke commented:
> Given the processor has 1GB of RAM, I feel the tradeoff for simply allocating the relatively small amount of memory would be a good tradeoff to the extra complexity for changing the buffer size.

Sounds good, how about 16K for the new size? Enough to call `screen.pixel()` 128x64 times without overflowing the queue and works out to a nice 1MiB of static allocation.

tehn commented:
Fantastic!
